test/workbox-background-sync/node/lib/test-Queue.mjs

```diff
@@ -0,0 +1,535 @@
+      });
+
+      const request = new Request('/');
+      await queue.addRequest(request);
+
+      expect(requestWillEnqueue.calledOnce).to.be.true;
+      expect(requestWillEnqueue.calledWith(sinon.match({
+        url: '/',
+        timestamp: sinon.match.number,
+        requestInit: sinon.match.object,
+      }))).to.be.true;
+    });
+  });
+
+  describe(`replayRequests`, function() {
+    it(`should try to re-fetch all requests in the queue`, async function() {
+      sandbox.spy(self, 'fetch');
+
+      const queue1 = new Queue('foo');
+      const queue2 = new Queue('bar');
+
+      // Add requests for both queues to ensure only the requests from
+      // the matching queue are replayed.
+      await queue1.addRequest(new Request('/one'));
+      await queue2.addRequest(new Request('/two'));
+      await queue1.addRequest(new Request('/three'));
+      await queue2.addRequest(new Request('/four'));
+      await queue1.addRequest(new Request('/five'));
+
+      await queue1.replayRequests();
+      expect(self.fetch.callCount).to.equal(3);
+
+      expect(self.fetch.getCall(0).calledWith(sinon.match({
+        url: '/one',
+      }))).to.be.true;
+
+      expect(self.fetch.getCall(1).calledWith(sinon.match({
+        url: '/three',
+      }))).to.be.true;
+
+      expect(self.fetch.getCall(2).calledWith(sinon.match({
+        url: '/five',
+      }))).to.be.true;
+
+      await queue2.replayRequests();
+      expect(self.fetch.callCount).to.equal(5);
+
+      expect(self.fetch.getCall(3).calledWith(sinon.match({
+        url: '/two',
+      }))).to.be.true;
+
+      expect(self.fetch.getCall(4).calledWith(sinon.match({
+        url: '/four',
+      }))).to.be.true;
+    });
+
+    it(`should remove requests after a successful retry`, async function() {
+      sandbox.spy(self, 'fetch');
+
+      const queue1 = new Queue('foo');
+      const queue2 = new Queue('bar');
+
+      // Add requests for both queues to ensure only the requests from
+      // the matching queue are replayed.
+      await queue1.addRequest(new Request('/one'));
+      await queue2.addRequest(new Request('/two'));
+      await queue1.addRequest(new Request('/three'));
+      await queue2.addRequest(new Request('/four'));
+      await queue1.addRequest(new Request('/five'));
+
+      await queue1.replayRequests();
+      expect(self.fetch.callCount).to.equal(3);
+
+      const db = await indexedDBHelper.getDB(
+          'workbox-background-sync', 'requests', {autoIncrement: true});
+
+      const itemsInObjectStore = [...await db.getAll()]
+          .map(([key, value]) => value.storableRequest.url);
+
+      expect(itemsInObjectStore.length).to.equal(2);
+      expect(itemsInObjectStore[0]).to.equal('/two');
+      expect(itemsInObjectStore[1]).to.equal('/four');
+    });
+
+    it(`should ignore (and remove) requests if maxRetentionTime has passed`,
+        async function() {
+      sandbox.spy(self, 'fetch');
+      const clock = sinon.useFakeTimers({
```

Comment:
Can you get this from the sandbox? If so you can drop the restore call at the end of this test.

Reply:
Wasn't in the documentation, but yep, appears to work. Done.